Microsoft.DotNet.Interactive.FSharp/FSharpKernel.fs

```diff
@@ -0,0 +1,96 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DotNet.Interactive.FSharp
+
+open System
+open System.ComponentModel
+open System.Diagnostics
+open System.Text
+open System.Threading.Tasks
+open Microsoft.DotNet.Interactive
+open Microsoft.DotNet.Interactive.Commands
+open Microsoft.DotNet.Interactive.Events
+
+type FSharpKernel() =
+    inherit KernelBase()
+    let mutable proc: Process = null
+    let write (s: string) = proc.StandardInput.Write(s)
+    let sentinelValue = Guid.NewGuid().ToString()
+    let sentinelFound = Event<_>()
+    let stdout = StringBuilder()
+    let waitForReady () =
+        async {
+            write <| sprintf ";;printfn \"\\n%s\";;\n" sentinelValue
+            do! Async.AwaitEvent sentinelFound.Publish
+        }
+    let startProcess () =
+        async {
+        if isNull proc then
+            let command = "dotnet"
```

Comment:
A couple of classes you could use to reduce some boilerplate here are `WorkspaceServer.Dotnet` (for finding the correct `dotnet` CLI) and `MLS.Agent.Tools.CommandLine` (for managing processes).

Reply:
`WorkspaceServer` has a dependency on `Microsoft.DotNet.Interactive.FSharp`, so I can't go the other way and use anything from that assembly. 😢

Reply:
Then it's probably appropriate to move the `Dotnet` class into a lower-level project.

Reply:
Right now it's pretty enmeshed in it's current location and it would require moving much more than just that single class (e.g., `MLS.Agent.Tools/(CommandLine|CommandLineResult)` and `WorkspaceServer/PackageSource`) as well as updating all of the references.

I'm all for refactoring this class out, but it feels beyond the scope of this PR.

Reply:
Spoke with @jonsequitur offline and we decided the proper thing to do was to move the `Dotnet` class down to the `MLS.Agent.Tools` assembly.  That's been done as of commit d5560b5053fc17e1e6434de4065cb0fe4be8239a.